src/webapi/date.rs

```diff
@@ -1,25 +1,1166 @@
+    #[inline]
+    fn set_milliseconds( &self, milliseconds_value: i32 ) {
+        js! { @(no_return)
+            @{self.as_ref()}.setMilliseconds(@{milliseconds_value});
+        };
+    }
+
+    /// The set_minutes() method sets the minutes for a specified date according to local time.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setMinutes)
+    #[inline]
+    fn set_minutes( &self, minutes: i32, _seconds: Option< i32 >, _milliseconds: Option< i32 > ) {
+        match _seconds {
+            Some(_seconds) => match _milliseconds {
+                Some(_milliseconds) => {
+                    let args_obj: HashMap< &str, Value > = [
+                        ("minutes", Value::Number( minutes.into())),
+                        ("seconds", Value::Number( _seconds.into())),
+                        ("ms",      Value::Number( _milliseconds.into()))
+                    ].iter().cloned().collect();
+
+                    js! { @(no_return)
+                        var args = @{args_obj};
+                        @{self.as_ref()}.setMinutes(args.minutes, args.seconds, args.ms);
+                    };
+                },
+                None => {
+                    js! { @(no_return)
+                        @{self.as_ref()}.setMinutes(@{minutes}, @{_seconds});
+                    };
+                },
+            },
+            None => {
+                js! { @(no_return)
+                    @{self.as_ref()}.setMinutes( @{minutes} );
+                };
+            }
+        }
+    }
+
+    /// The set_month() method sets the month for a specified date according to the currently set
+    /// year.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setMonth)
+    #[inline]
+    fn set_month( &self, month: i32, _day: Option< i32 > ) {
+        match _day {
+            Some(_day) => {
+                js! { @(no_return)
+                    @{self.as_ref()}.setMonth(@{month}, @{_day});
+                };
+            },
+            None => {
+                js! { @(no_return)
+                    @{self.as_ref()}.setMonth(@{month});
+                };
+            }
+        }
+    }
+
+    /// The set_seconds() method sets the seconds for a specified date according to local time.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/setSeconds)
+    #[inline]
+    fn set_seconds( &self, seconds: i32, _ms: Option< i32 > ) {
+        match _ms {
+            Some(_ms) => {
+                js! { @(no_return)
+                    @{self.as_ref()}.setSeconds(@{seconds}, @{_ms});
+                };
+            },
+            None => {
+                js! { @(no_return)
+                    @{self.as_ref()}.setSeconds(@{seconds});
+                };
+            }
+        }
```

Comment:
Hmmmm... I'm really conflicted on how we should handle methods like this. I **really** don't like the optional arguments, and after I took a better look at the APIs they seem to be mostly unnecessary. For example, if you want to set the milliseconds you can either use `setSeconds(..., ms)` or `setMilliseconds(ms)` - two methods which do the same thing, it's just that `setSeconds` also sets the seconds at the same time.

So how about we have these methods do only one thing? (e.g. `setSeconds` only sets the seconds, etc.)

Reply:
definitely had the same thought while implementing this: setters should set what they say they're setting...but followed the documentation through. I'd be happy to make these single arg functions